File: backend/src/entities/service_provider.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "service_providers")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub provider_name: String,
    pub provider_code: String,
    pub short_name: String,
    pub logo_url: Option<String>,
    pub contact_person: String,
    pub contact_phone: String,
    pub contact_email: String,
    pub headquarters: String,
    pub service_area: String,
    pub business_license: String,
    pub remarks: Option<String>,
    pub status: String,
    pub created_at: String,
    pub updated_at: Option<String>,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(has_many = "super::machine_room::Entity")]
    MachineRoom,
    #[sea_orm(has_many = "super::cloud_platform_config::Entity")]
    CloudPlatformConfig,
}

impl Related<super::machine_room::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::MachineRoom.def()
    }
}

impl Related<super::cloud_platform_config::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::CloudPlatformConfig.def()
    }
}

impl ActiveModelBehavior for ActiveModel {}

File: apps/web/apps/web-antd/src/api/scan/service-provider.ts
import { requestClient } from '#/api/request';

export namespace ScanServiceProviderApi {
  export interface ServiceProvider {
    id?: number; provider_name: string; provider_code: string;
    short_name: string; logo_url?: string; contact_person: string;
    contact_phone: string; contact_email: string; headquarters: string;
    service_area: string; business_license: string; remarks?: string;
    status: string; created_at: string; updated_at?: string;
  }
}
export function getServiceProviderList() { return requestClient.get<ScanServiceProviderApi.ServiceProvider[]>('/asset-ops/service-providers'); }
export function getServiceProvider(id: number) { return requestClient.get<ScanServiceProviderApi.ServiceProvider>(`/asset-ops/service-providers/${id}`); }
export function createServiceProvider(data: ScanServiceProviderApi.ServiceProvider) { return requestClient.post('/asset-ops/service-providers', data); }
export function updateServiceProvider(id: number, data: ScanServiceProviderApi.ServiceProvider) { return requestClient.put(`/asset-ops/service-providers/${id}`, data); }
export function deleteServiceProvider(id: number) { return requestClient.delete(`/asset-ops/service-providers/${id}`); }
